src/helpers/slack.ts

```diff
@@ -10,29 +11,59 @@ import {
 const fs = require("fs");
 
 export function getEmojisToReactWith(text: string): Array<string> {
-    const emojisToReactWith: Array<string> = [];
+    const lowerCaseText = text.toLowerCase();
 
     // Read list of emojis from file
-    const rawdata = fs.readFileSync(
-        process.env.EMOJIS_FILE_PATH || "./helpers/files/emojis.json"
-    );
-    const emojis: Record<string, Array<string>> = JSON.parse(rawdata);
+    const emojis: Record<string, Array<string>> = readEmojisFromFile();
 
-    // flattens the array of keywords
     const keywords = Object.values(emojis).flat();
 
     // search for each keyword in the text
+    const emojisToReactWith = findMatchingEmojiKeywords(
+        keywords,
+        lowerCaseText,
+        emojis
+    );
+
+    return Array.from(new Set(emojisToReactWith));
+}
+
+function readEmojisFromFile(): Record<string, Array<string>> {
+    var emojisFilePath: string;
+    if (process.env.EMOJIS_FILE_PATH) {
+        // check if the file path is valid
+        if (!fs.existsSync(process.env.EMOJIS_FILE_PATH)) {
+            console.error(
+                `Emojis file path ${process.env.EMOJIS_FILE_PATH} is invalid.`
+            );
+            return {};
```

Comment:
[Non blocking comment]

This function should only read data and nothing more as per SRP. May be move the `if (process.env.EMOJIS_FILE_PATH)` and `fs.existsSync` to a reusable utility function. eg: `IsValidEmojiFile()`

Reply:
I have created a separate function to get emojis file path, that way the function is only checking if the function returned a file path and then reading from the file